addons/flannel/0.21.5/install.sh

```diff
@@ -270,9 +270,35 @@ function remove_weave() {
 
     # all nodes
     ip link delete weave
+
     rm -rf /var/lib/weave
     rm -rf /etc/cni/net.d/*weave*
     rm -rf /opt/cni/bin/weave*
+
+    logStep "Verifying Weave removal"
+    # Check if any Weave-related resources are still present
+    weave_resources=$(kubectl get all --all-namespaces | grep weave || true)
+
+    if [ -n "$weave_resources" ]; then
+        logWarn "Weave resources are still present. Deleting Weave resources with grace period up to 3 minutes..."
+        kubectl -n kube-system delete $(kubectl get all -n kube-system | grep weave | awk '{print $1}') --grace-period=180
+
+        # Check for remaining resources
+        weave_resources=$(kubectl get all --all-namespaces | grep weave || true)
```

Comment:
```suggestion
        weave_resources=$(kubectl -n kube-system get pods --no-headers -l name=weave-net)
```

Reply:
We cannot do that. See that we need to ensure that all kind of resources deleted above were removed and not only Pods. 
    

> kubectl -n kube-system delete daemonset weave-net
>     kubectl -n kube-system delete rolebinding weave-net
>     kubectl -n kube-system delete role weave-net
>     kubectl delete clusterrolebinding weave-net
>     kubectl delete clusterrole weave-net
>     kubectl -n kube-system delete serviceaccount weave-net
>     kubectl -n kube-system delete secret weave-passwd